ssl/ssl_lib.c

```diff
@@ -6770,26 +6769,25 @@ static int nss_keylog_int(const char *prefix,
     if ((out = cursor = OPENSSL_malloc(out_len)) == NULL)
         return 0;
 
-    strcpy(cursor, prefix);
+    memcpy(cursor, prefix, prefix_len);
     cursor += prefix_len;
     *cursor++ = ' ';
 
-    for (i = 0; i < parameter_1_len; i++) {
-        sprintf(cursor, "%02x", parameter_1[i]);
```

Comment:
was there a problem with using sprintf here?  Seems like a simpler solution in this code

Reply:
This breaks a no-stdio build and there is no sprintf call from libssl.
I may replace it by BIO_snprintf ?

Reply:
I am in favour of removing calls to `sprintf()` as they often fall foul of "do not use function X" security API bans, as it is so easy to mis-use (it's use is fine here, though).

To avoid duplication of code, how about a small function `static size_t to_hex(char *buf, uint8_t n)` that writes `n` in hex to `buf` and returns the number of bytes written (2)? Then `cursor += to_hex(cursor, parameter_1[i])`

Reply:
I support a to_hex function/macro/inline.
We've way too many instances of things like this being duplicated all over the place.

Reply:
I factorize these "to_hex" conversion code as far as I can.

@tom-cosgrove-arm  please take a look.

Reply:
point resolved, i think.
Please review.